### DISCUSSION TITLE
Add toggle service to climate

### DISCUSSION THREAD

**Discussion by arturpragacz** [home-assistant/architecture#967] [Entity Models]
This is related to https://github.com/home-assistant/core/pull/100418.

## Proposal

Climate already has both TURN_ON and TURN_OFF services. Adding TOGGLE as well makes it more consistent. It breaks default user expectations when something can be both turned on and turned off, but cannot be toggled.

## Alternatives

One could argue that an alternative to adding toggle would be to remove turn_on and turn_off instead. This is worse for several reasons.
- They were implemented over 4 years ago. Removing them now would be a substantial breaking change.
- They are useful when one wants to for example turn off many disparate devices with one call. It can be done by calling homeassistant.turn_off. Similarly for turn_on and toggle.
- Some external programs that integrate with Home Assistant don't have good support for all entity types, but they almost always support the generic homeassistant.{turn_on,turn_off,toggle}. Supporting these services as broadly as possible is therefore good for interoperability.

**Comment by gjohansson-ST** [comment]
I think this makes sense as in many devices there is a power button so you actually toggle it and not necessarily press on or off.

**Reply by gjohansson-ST** [reply]
Implemented in https://github.com/home-assistant/core/pull/100418

**Comment by emontnemery** [comment]
This is a good suggestion, a PR implementing this is welcome 👍
